Objects/Converters/ConverterRevit/ConverterRevitShared/ConversionUtils.cs

```diff
@@ -251,19 +251,30 @@ public void GetAllRevitParamsAndIds(Base speckleElement, DB.Element revitElement
 
       speckleElement["worksetId"] = revitElement.WorksetId.ToString();
 
-      // for the category, we should be mirroring how we handle parsing of built-in-categories for revit 2023/2024
-      // this is because different built-in-categories may have the same name, eg "OST_Railings" and "OST_StairsRailing" both have a Category name of "Railing"
+      // assign the category *if* it hasn't already been assigned (eg, DirectShape conversion should already have assigned the category).
       var category = revitElement.Category;
-      if (category is not null)
+      var speckleCategory = speckleElement["category"];
+      if (speckleCategory is null && category is not null)
       {
-        var categoryName = revitElement.Category.Name;
-#if !(REVIT2020 || REVIT2021 || REVIT2022)
-        if (Categories.GetRevitCategoryFromBuiltInCategory(category.BuiltInCategory, out RevitCategory c))
+        var categoryName = category.Name;
+        switch (speckleElement["category"])
         {
-          categoryName = c.ToString();
-        }
+          case RevitCategory: // DirectShape as of 2.16 is the only class with `category` prop of type `RevitCategory`
+            speckleElement["category"] = Categories.GetSchemaBuilderCategoryFromBuiltIn(categoryName);
+            break;
+
+          case string:
```

Comment:
We actually don't need this switch. The RevitCategory type is an enum which means it's a non-nullable reference type. So the above statement, `if (speckleCategory is null)` means that this switch case will never hit.

Reply:
Yeah I considered this, but included the switch because it makes explicit that we have different types - if we change the logic in the future to handle cases of incorrect default category assignment (we don't currently have a `none` enum, so the 0 value is `abutment` currently), the null check would be removed and this would prevent us from going through this again

Reply:
Hmm, okay. Not sure how I feel about including code that is unreachable. Especially given that it sounds like @AlanRynne and @JR-Morgan have a plan to change the direct shape category from an enum to a string, I don't think we should include this unreachable code just to be explicit that the category can be an enum (when it shouldn't be moving forward). Happy to hear your thoughts on this

Reply:
If anything I would prefer changing `category` to be it's own `RevitCategory:Category` class, so we can store display names as well as built-in names for revit 😭

Reply:
Removing the switch in favor of an inline comment for now, in case we do need to make category re-assignments prior to the direct shape category type change